File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/PagingAndSortingDto.java
package uk.gov.justice.digital.hmpps.riskprofiler.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class PagingAndSortingDto {
    public static final String HEADER_PAGE_OFFSET = "Page-Offset";
    public static final String HEADER_PAGE_LIMIT = "Page-Limit";
    public static final String HEADER_TOTAL_RECORDS = "Total-Records";
    public static final String HEADER_SORT_FIELDS = "Sort-Fields";
    public static final String HEADER_SORT_ORDER = "Sort-Order";

    private Long pageOffset;
    private Long pageLimit;
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/model/PagingAndSortingDto.kt
package uk.gov.justice.digital.hmpps.riskprofiler.model

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
data class PagingAndSortingDto(
  val pageOffset: Long? = null,
  val pageLimit: Long? = null,
) {
  constructor() : this(null, null)

  companion object {
    const val HEADER_PAGE_OFFSET = "Page-Offset"
    const val HEADER_PAGE_LIMIT = "Page-Limit"
    const val HEADER_TOTAL_RECORDS = "Total-Records"
    const val HEADER_SORT_FIELDS = "Sort-Fields"
    const val HEADER_SORT_ORDER = "Sort-Order"
  }
}
